roles/ssh-manager/tasks/process_user.yml

```diff
@@ -0,0 +1,47 @@
+---
+- name: "Process SSH keys for {{ unix_user }}"
+  block:
+    - name: "Get user's home directory for {{ unix_user }}"
+      ansible.builtin.set_fact:
+        user_home: "{{ getent_passwd_result.ansible_facts.getent_passwd[unix_user][4] }}"
+
+    - name: "Initialize empty key list for {{ unix_user }}"
+      ansible.builtin.set_fact:
+        user_ssh_keys: []
+
+    - name: "Build list of SSH keys for {{ unix_user }}"
+      ansible.builtin.set_fact:
+        user_ssh_keys: "{{ (user_ssh_keys + person_keys) | unique }}"
+      loop: "{{ ssh_manager_authorized_persons | dict2items }}"
+      loop_control:
+        loop_var: person_entry
+        label: "{{ person_entry.key }}"
+      vars:
+        person_name: "{{ person_entry.key }}"
+        person_data: "{{ person_entry.value }}"
+        # this will match if the person is authorized for all machines or the current machine
+        machine_matches: "{{ inventory_hostname in query('inventory_hostnames', person_data['machines'] if person_data['machines'] is string else person_data['machines'] | join(',')) }}"
+        user_matches: "{{ unix_user in person_data['users'] }}"
+        person_keys: "{{ person_data['keys'] }}"
```

Comment:
(reply to @kysrpex  https://github.com/usegalaxy-eu/infrastructure-playbook/pull/1673/files#r2486631829)
Maybe we can only collect the persons authorized for the user, so the template can use the whole person dictionary
~~~
{% for person in user_authorized_persons %}
{% for pub_key in person.keys %}
{{ pub_key }} {{ person.key }}
~~~
(`key` and `keys` is a bit difficult to distinguish, what I am referring to here is python's `key()` method)